internal/services/storage/storage_account_data_source.go

```diff
@@ -368,6 +378,23 @@ func dataSourceStorageAccountRead(d *pluginsdk.ResourceData, meta interface{}) e
 			}
 			d.Set("secondary_blob_connection_string", secondaryBlobConnectStr)
 		}
+
+		// Setting the encryption key type to "Service" in PUT. The following GET will not return the queue/table in the service list of its response.
+		// So defaults to setting the encryption key type to "Service" if it is absent in the GET response. Also, define the default value as "Service" in the schema.
+		var (
+			queueEncryptionKeyType = "Service"
+			tableEncryptionKeyType = "Service"
```

Comment:
can we use the constants here:

```suggestion
			queueEncryptionKeyType = string(storage.KeyTypeService)
			tableEncryptionKeyType = string(storage.KeyTypeService)
```